### PR TITLE
[FLINK-18117][e2e] Add debugging information for hadoop startup

### PR DESCRIPTION
**Pull request by rmetzger**
## What is the purpose of the change

This adds some non-invasive environment debugging to understand why the namenode can not allocate the http port inside a docker container.
I was not able to reproduce the failing e2e test.

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh [hunk 1]
```diff
@@ -92,7 +92,12 @@ if [ "$1" == "--help" -o "$1" == "-h" ]; then
     exit 0
 elif [ "$1" == "master" ]; then
     yes| sudo -E -u hdfs $HADOOP_PREFIX/bin/hdfs namenode -format
-
+    sudo -E sudo netstat -tulpn >> /var/log/hadoop/debugging.out
+    echo "--------------------" >> /var/log/hadoop/debugging.out
+    sudo -E sudo ps -aux >> /var/log/hadoop/debugging.out
+    echo "--------------------" >> /var/log/hadoop/debugging.out
+    sudo -E sudo jps -v >> /var/log/hadoop/debugging.out
```
Comment (XComp) [inline flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh:99]: Are you sure that you want to use `sudo -E sudo #...`? I couldn't test it quickly but it looks strange.

File: flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh [hunk 2]
```diff
@@ -92,7 +92,12 @@ if [ "$1" == "--help" -o "$1" == "-h" ]; then
     exit 0
 elif [ "$1" == "master" ]; then
     yes| sudo -E -u hdfs $HADOOP_PREFIX/bin/hdfs namenode -format
-
+    sudo -E sudo netstat -tulpn >> /var/log/hadoop/debugging.out
```
Comment (XComp) [inline flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh:95]: Just as a side-note: I usually added also `-e` as a parameter getting "extended" output. Not sure whether it helps in that case but it's just nice to read for someone who knows German: `netstat -tulpen` 😄

**Review by XComp — APPROVED** [review]
One thing I am not sure about. But it looks good otherwise. 👍

File: flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh (hunk 2 above)
Comment (rmetzger) [reply flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh:95]: I tried the -e argument, but it only adds things like "package count" per port, which is not relevant for understanding who's allocating the port. (I copy pasted "tulpn" from somewhere, and also remembered the -e)

File: flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh (hunk 1 above)
Comment (rmetzger) [reply flink-end-to-end-tests/test-scripts/docker-hadoop-secure-cluster/bootstrap.sh:99]: oh, that's clearly a mistake. Good catch! I'll fix it.

**Comment by rmetzger** [comment]
Thanks a lot for your review. Merging ...

**Pull request merged**